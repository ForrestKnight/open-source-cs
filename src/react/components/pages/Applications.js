/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import ApplicationsCourses from '../../data/ApplicationsCourses';


/* Component */
export class Applications extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Applications');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Applications" className="page">
                <CourseSection 
                    name="Applications"
                    courses={ApplicationsCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Applications);