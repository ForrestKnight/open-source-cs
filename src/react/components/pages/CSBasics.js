/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import CSBasicsCourses from '../../data/CSBasicsCourses';


/* Component */
export class CSBasics extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('CSBasics');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="CSBasics" className="page">
                <CourseSection 
                    name="CS Basics"
                    courses={CSBasicsCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(CSBasics);