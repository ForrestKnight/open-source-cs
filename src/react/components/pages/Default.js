/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Component */
export class Default extends React.Component {
    constructor(props) {
        super(props);
    };


    componentWillMount() {
        this.page_ID__Set('Default');
    };


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    };


    render() {
        return (
            <div id="Default">
                <section className="Default__header">
                    <h2>Redux</h2>
                    <h2>Boilerplate.</h2>
                    <h4>By Robby Bergers</h4>
                </section>

                <section className="Default__content">
                    <article className="Default__links">
                        <a 
                            className="Default__link"
                            href="https://github.com/RobbyB97/JavascriptReactBoilerplate"
                        >
                            View on GitHub
                        </a>

                        <a
                            className="Default__link"
                            href="https://paypal.me/robbybobby123"
                        >
                            Donate
                        </a>
                    </article>

                    <p className="Default__tagline">
                        A better create-react-app.
                    </p>

                    <article>
                        <p>
                            Looking to create a React app? Hit the ground running with redux, sass, source mapping, webpack, node, express, and jest unit testing. This template also comes with a page loader built in, wildcard route, responsive navigation UI and an easily changeable color scheme.
                        </p>
                    </article>
                    
                    <article className="Default__donate">
                        <p>
                            Creating software such as this boilerplate takes a lot of time, effort, attention to detail, and the occasional excellent anger management. If you found this boilerplate helpful, please consider <a href="https://paypal.me/robbybobby123">donating</a>.
                        </p>
                    </article>
                </section>
            </div>
        );
    };
};


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Default);