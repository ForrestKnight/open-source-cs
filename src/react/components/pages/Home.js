/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import Description from '../sections/Description';
import CourseSection from '../sections/CourseSection';


/* Data */
import CSBasicsCourses from '../../data/CSBasicsCourses';
import ProgrammingCourses from '../../data/ProgrammingCourses';
import MathCourses from '../../data/MathCourses';
import SystemsCourses from '../../data/SystemsCourses';


/* Component */
export class Home extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Home');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Home" className="Home">
                <Description />
                
                <CourseSection 
                    name="CS Basics"
                    courses={CSBasicsCourses}
                />

                <CourseSection 
                    name="Programming"
                    courses={ProgrammingCourses}
                />

                <CourseSection 
                    name="Math"
                    courses={MathCourses}
                />

                <CourseSection 
                    name="Systems"
                    courses={SystemsCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Home);