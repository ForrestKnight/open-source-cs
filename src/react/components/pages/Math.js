/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import MathCourses from '../../data/MathCourses';


/* Component */
export class Math extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Math');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Math" className="page">
                <CourseSection 
                    name="Math"
                    courses={MathCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Math);