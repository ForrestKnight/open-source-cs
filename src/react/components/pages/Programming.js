/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import ProgrammingCourses from '../../data/ProgrammingCourses';


/* Component */
export class Programming extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Programming');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Programming" className="page">
                <CourseSection 
                    name="Programming"
                    courses={ProgrammingCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Programming);