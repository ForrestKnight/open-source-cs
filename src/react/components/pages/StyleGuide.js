/* Packages */
import React from 'react';
import {connect} from 'react-redux';
import {NavLink} from 'react-router-dom';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Component */
export class StyleGuide extends React.Component {
    constructor(props) {
        super(props);
    };


    componentWillMount() {
        this.page_ID__Set('StyleGuide');
    };


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    };


    render() {
        return (
            <div id="StyleGuide">
                <section className="StyleGuide__headers">
                    <h1>This is an h1 tag</h1>
                    <h2>This is an h2 tag</h2>
                    <h3>This is an h3 tag</h3>
                    <h4>This is an h4 tag</h4>
                    <h5>This is an h5 tag</h5>
                    <h6>This is an h6 tag</h6>
                </section>

                <section className="StyleGuide__normalText">
                    <p>This is a normal p tag</p>
                    <p>This is a p tag <i>with an i tag</i></p>
                    <p>This is a p tag <em>with an em tag</em></p>
                    <p>This is a p tag <b>with a b tag</b></p>
                    <p>This is a p tag <mark>with a mark tag</mark></p>
                    <p>This is a p tag <span>with a span tag</span></p>
                    <p>This is a p tag <small>with a small tag</small></p>
                    <p>This is a p tag <strong>with a strong tag</strong></p>
                    <p>This is a p tag <u>with a u tag</u></p>
                    <p>This is a p tag <NavLink to="/StyleGuide">with an a/navlink tag</NavLink></p>
                    <p>This is a p tag <code>with a code tag</code></p>
                    <p>This is a p tag <cite>with a cite tag</cite></p>
                </section>

                <section className="StyleGuide__button">
                    <button onClick={(e) => e.preventDefault}>This is a button tag</button><br/>
                </section>

                <section className="StyleGuide__list">
                    <ol>
                        <li>This</li>
                        <li>is</li>
                        <li>an</li>
                        <li>ordered</li>
                        <li>list</li>
                    </ol>

                    <ul>
                        <li>This</li>
                        <li>is</li>
                        <li>an</li>
                        <li>unordered</li>
                        <li>list</li>
                    </ul>
                </section>
            </div>
        );
    };
};


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(StyleGuide);