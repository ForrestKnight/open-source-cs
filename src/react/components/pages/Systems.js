/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import SystemsCourses from '../../data/SystemsCourses';


/* Component */
export class Systems extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Systems');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Systems" className="page">
                <CourseSection 
                    name="Systems"
                    courses={SystemsCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Systems);