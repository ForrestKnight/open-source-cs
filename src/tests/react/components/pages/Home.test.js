/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {Home} from '../../../../react/components/pages/Home';


/* Test Config */
let component, page_ID__Set;
beforeEach(() => {
    page_ID__Set = jest.fn();
    component = shallow(
        <Home
            page_ID__Set={page_ID__Set}
        />
    );
});


/* Tests */
test('Render Home page', () => {
    expect(component).toMatchSnapshot();
});


test('Page_ID__Set function', () => {
    setTimeout(() => {
        expect(page_ID__Set).toHaveBeenCalled();
    }, 500);
});

