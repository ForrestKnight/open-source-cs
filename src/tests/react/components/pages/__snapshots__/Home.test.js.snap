// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home page 1`] = `
<div
  className="Home"
  id="Home"
>
  <Description />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "10",
          "effort": "10-20",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/cs50s-introduction-to-computer-science",
            "text": "Intro to Computer Science",
          },
          "school": "Harvard",
        },
        Object {
          "duration": "6",
          "effort": "2-5",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/what-is-a-proof",
            "text": "Mathematical Thinking in Computer Science",
          },
          "school": "UC San Diego",
        },
      ]
    }
    name="CS Basics"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "4",
          "effort": "4-8",
          "frequency": "Twice a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/java-programming",
            "text": "Java Programming: Solving Problems with Software",
          },
          "school": "Duke",
        },
        Object {
          "duration": "4",
          "effort": "4-8",
          "frequency": "Twice a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/java-programming-arrays-lists-data",
            "text": "Java Programming: Arrays, Lists, and Structured Data",
          },
          "prerequisites": Array [
            "Java Programming: Solving Problems with Software",
          ],
          "school": "Duke",
        },
        Object {
          "duration": "6",
          "effort": "4-6",
          "frequency": "Every week",
          "name": Object {
            "link": "https://www.coursera.org/learn/object-oriented-java",
            "text": "Object Oriented Programming in Java",
          },
          "prerequisites": Array [
            "Java Programming: Arrays, Lists, and Structured Data",
          ],
          "school": "Duke",
        },
        Object {
          "duration": "6",
          "effort": "6-10",
          "frequency": "Every week",
          "name": Object {
            "link": "https://www.coursera.org/learn/data-structures-optimizing-performance",
            "text": "Data Structures and Performance",
          },
          "prerequisites": Array [
            "Object Oriented Programming in Java",
          ],
          "school": "Duke",
        },
        Object {
          "duration": "4",
          "effort": "4-8",
          "frequency": "Twice a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/java-programming-design-principles",
            "text": "Java Programming: Principles of Software Design",
          },
          "prerequisites": Array [
            "Java Programming: Arrays, Lists, and Structured Data",
          ],
          "school": "Duke",
        },
        Object {
          "duration": "4",
          "effort": "3-6",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/java-programming-recommender",
            "text": "Java Programming: Build a Recommendation System",
          },
          "prerequisites": Array [
            "Java Programming: Principles of Software Design",
          ],
          "school": "Duke",
        },
        Object {
          "duration": "5",
          "effort": "8-16",
          "frequency": "Once a month",
          "name": Object {
            "link": "coursera.org/learn/programming-languages",
            "text": "Programming Languages, Part A",
          },
          "prerequisites": Array [
            "Object Oriented Programming in Java",
          ],
          "school": "UW",
        },
        Object {
          "duration": "3",
          "effort": "8-16",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/programming-languages-part-b",
            "text": "Programming Languages, Part B",
          },
          "prerequisites": Array [
            "Programming Languages, Part A",
          ],
          "school": "UW",
        },
        Object {
          "duration": "3",
          "effort": "8-16",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/programming-languages-part-c",
            "text": "Programming Languages, Part C",
          },
          "prerequisites": Array [
            "Programming Languages, Part B",
          ],
          "school": "UW",
        },
      ]
    }
    name="Programming"
  />
</div>
`;
